### PR TITLE
More logging during training and scoring on all prepared image data

### PR DESCRIPTION
**Pull request by fujikosu**
@noodlefrenzy  Could you review these changes?

This PR adds accuracy logging to know more detailed information during training in `train_keras.py`.
All of added logs go to AML logging now for reviewing the result later with more useful data in `train_keras.py`.
The implementation of top n scoring gets simplified by using `top_k_categorical_accuracy` in `score_keras.py`.
This PR makes scoring script use all files in testing directory because using all prepared test images are what I expected as default behavior in `score_keras.py`.

### CODE REVIEW

File: train_keras.py [hunk 1]
```diff
@@ -312,7 +311,11 @@ def train_model(img_path,
             workers=cpu_count,
             callbacks=callbacks)
         execution_time = time.perf_counter() - start_time
-        aml_run_logger.log("Initial training execution time", execution_time)
+    aml_run_logger.log("Initial training execution time", execution_time)
```
Comment (noodlefrenzy) [inline train_keras.py:314]: Much nicer

File: score_keras.py [hunk 2]
```diff
@@ -54,17 +57,18 @@ def load_model_and_images(model_file, image_dir, seed):
...
 
     return model, test_gen, classes
 
-def evaluate_model(model, image_gen, classes, num_batches, output_metrics, output_image, top_n=None, beta=1.):
-    for batch_num in range(num_batches):
-        logger.info('Processing batch %d of %d' % (batch_num, num_batches))
+def evaluate_model(model, image_gen, classes, output_metrics, output_image, top_n=None, beta=1.):
+    # Iterate all batches for testing
+    for batch_num in range(len(image_gen)):
+        logger.info('Processing batch %d of %d' % (batch_num, len(image_gen)))
```
Comment (noodlefrenzy) [inline score_keras.py:71]: Change `Processing batch` to `Processing image`

**Review by noodlefrenzy — APPROVED** [review]
:shipit: I like these changes. Will almost certainly integrate some of them back into my main repo.

File: score_keras.py (hunk 2 above)
Comment (fujikosu) [reply score_keras.py:71]: Do you mean current logs below don't give you enough information?
```
Found 1064 images belonging to 19 classes.
Processing batch 0 of 34
Processing batch 1 of 34
```
Maybe this would be more helpful?
```
Found 1064 images belonging to 19 classes.
Batch size is 32
Processing batch 0 of 34
Processing batch 1 of 34
```

Comment (noodlefrenzy) [reply score_keras.py:71]: No, perhaps I'm misunderstanding the generator. It looked to me like you were processing the images one at a time. If so, just wanted you to change the word "batch" to "image" to reflect that.
